pydantic_xml/serializers/factories/primitive.py

```diff
@@ -67,8 +67,8 @@ class AttributeSerializer(Serializer):
     def from_core_schema(cls, schema: PrimitiveTypeSchema, ctx: Serializer.Context) -> 'AttributeSerializer':
         namespaced_attrs = ctx.namespaced_attrs
         name = ctx.entity_path or ctx.field_alias or ctx.field_name
-        ns = ctx.entity_ns or (ctx.parent_ns if namespaced_attrs else None)
-        nsmap = merge_nsmaps(ctx.entity_nsmap, ctx.parent_nsmap)
+        ns = select_ns(ctx.entity_ns, ctx.parent_ns if namespaced_attrs else None)
+        nsmap = cls._merge_attr_nsmaps(ctx.entity_nsmap, ctx.parent_nsmap)
```

Comment:
using select ns broke this [test](https://github.com/dapper91/pydantic-xml/blob/master/tests/test_namespaces.py#L43)

I believe it is because of this XML attribute rule: https://stackoverflow.com/questions/41561/xml-namespaces-and-attributes

from my reading it seems if the element has no defined namespace (default namespace is not a defined namespace) then the attribute belongs to [NO namespace](https://stackoverflow.com/questions/41561/xml-namespaces-and-attributes#comment131787005_46865). 

So the parsed qualified name for `submodel` `attr1`  is actually `attr1`. But this change to `select_ns` will choose the ns to be `''` and make the qualified name `{http://test3.org}attr1` as it gets `http://test3.org` from `nsmap`

To fix this issue I've decided to pop the default ns `''` from nsmap so that it resolves to `attr1` as expected.

Is there a better way to do this?

Reply:
Thanks for the info.
I think the test is incorrect. I suppose it is better to fix the test. 
Let's remove the namespace from the `attr1` at all:

```python
    class TestSubModel(BaseXmlModel, tag='submodel', ns='', nsmap={'': 'http://test3.org', 'tst': 'http://test4.org'}):
        attr1: int = attr()
        attr2: int = attr(ns='tst')
        element1: str = element(ns='')
```

Reply:
Thanks for the info.
I think the test is incorrect. I suppose it is better to fix the test. 
Let's remove the namespace from the `attr1` at all:

```python
    class TestSubModel(BaseXmlModel, tag='submodel', ns='', nsmap={'': 'http://test3.org', 'tst': 'http://test4.org'}):
        attr1: int = attr()
        attr2: int = attr(ns='tst')
        element1: str = element(ns='')
```

and raise an exception if an attribute's namespace is '', smth like this:

```python
raise errors.ModelFieldError(ctx.model_name, ctx.field_name,"attributes with default namespace are forbidden")
```

I think it will be more obvious than just dropping namespace from the nsmap.

Reply:
👍 I agree that sounds better. I will make those changes